www/js/control/AppStatusModal.tsx

```diff
@@ -0,0 +1,458 @@
+            console.log("fix and refresh battery optimization permissions");
+            return checkOrFix(ignoreBatteryOptCheck, window['cordova'].plugins.BEMDataCollection.fixIgnoreBatteryOptimizations,
+                true);
+        };
+        let checkBatteryOpt = function() {
+            console.log("fix and refresh battery optimization permissions");
+            return checkOrFix(ignoreBatteryOptCheck, window['cordova'].plugins.BEMDataCollection.isIgnoreBatteryOptimizations,
+                false);
+        };
+        var androidUnusedDescTag = "intro.appstatus.unusedapprestrict.description.android-disable-gte-12";
+        if (osver < 12) {
+            androidUnusedDescTag= "intro.appstatus.unusedapprestrict.description.android-disable-lt-12";
+        }
+        let unusedAppsUnrestrictedCheck = {
+            name: t("intro.appstatus.unusedapprestrict.name"),
+            desc: t(androidUnusedDescTag),
+            fix: fixPerms,
+            refresh: checkPerms
+        }
+        let ignoreBatteryOptCheck = {
+            name: t("intro.appstatus.ignorebatteryopt.name"),
+            desc: t("intro.appstatus.ignorebatteryopt.description.android-disable"),
+            fix: fixBatteryOpt,
+            refresh: checkBatteryOpt
+        }
+        let tempChecks = checkList;
+        tempChecks.push(unusedAppsUnrestrictedCheck, ignoreBatteryOptCheck);
+        setCheckList(tempChecks);
+    }
+
+    function setupPermissionText() {
+        let tempExplanations = explanationList;
+
+        let overallFitnessName = t('intro.appstatus.overall-fitness-name-android');
+        let locExplanation = t('intro.appstatus.overall-loc-description');
+        if(platform == "ios") {
+            overallFitnessName = t('intro.appstatus.overall-fitness-name-ios');
+            if(osver < 13) {
+                locExplanation = (t("intro.permissions.locationPermExplanation-ios-lt-13"));
+            } else {
+                locExplanation = (t("intro.permissions.locationPermExplanation-ios-gte-13"));
+            }
+        }
+        tempExplanations.push({name: t('intro.appstatus.overall-loc-name'), desc: locExplanation});
+        tempExplanations.push({name: overallFitnessName, desc: t('intro.appstatus.overall-fitness-description')});
+        tempExplanations.push({name: t('intro.appstatus.overall-notification-name'), desc: t('intro.appstatus.overall-notification-description')});
+        tempExplanations.push({name: t('intro.appstatus.overall-background-restrictions-name'), desc: t('intro.appstatus.overall-background-restrictions-description')});
+
+        setExplanationList(tempExplanations);
+  
+        //waiting on samsung feedback, need more information
+        setBackgroundRestricted(false);
+        if(window['device'].manufacturer.toLowerCase() == "samsung") {
+          setBackgroundRestricted(true);
+          setAllowBackgroundInstructions(t("intro.allow_background.samsung"));
+        }
+  
```

Comment:
so these are not actually used right now pending "dealing with Samsung", correct?

Reply:
Correct, this is pending update based on ["dealing with Samsung"](https://github.com/e-mission/e-mission-docs/issues/535)